rs/replicated_state/src/canister_state/queues.rs

```diff
@@ -702,12 +702,14 @@ impl CanisterQueues {
     ///
     ///  * If this is a guaranteed `Response`, the protocol should have reserved a
     ///    slot for it, so the push should not fail for lack of one (although an
-    ///    error may be returned in case of a bug in the upper layers).
+    ///    error may be returned in case of a bug in the upper layers) and `Ok(true)`
+    ///    is returned.
     ///  * If this is a best-effort `Response`, a slot is available and no duplicate
-    ///    (time out) response is already enqueued, it is enqueued.
+    ///    (time out) response is already enqueued, it is enqueued and `Ok(true)` is
+    ///    returned.
     ///  * If this is a best-effort `Response` and a duplicate (time out) response
     ///    is already enqueued (which is implicitly true when no slot is available),
-    ///    the response is silently dropped and `Ok(())` is returned.
+    ///    the response is silently dropped and `Ok(false)` is returned.
```

Comment:
I believe that the `Ok(false)` addition is fine. But Instead of modifying the first two bullet points, it's better to explain the `Ok(true)` / `Ok(false)` meaning either at the top or under a `Return` heading (likely replacing `Errors`) below.

As is, this bullet point list only covers responses, it's unclear what an `Ok(false)` would mean (or if it's possible at all) for a request.

OTOH, the changes may be OK as they are if you add an `Ok(true)` above to the paragraph talking about requests. AND an explanation of the meaning of the boolean flag, somewhere or other.

Reply:
I added `Ok(true)` for requests. `Ok(false)` for requests is not possible because we don't silently drop requests.
I still think it might be cleaner to have separate paths for inducting requests or responses since `bool` only makes sense for responses really and other reasons that have nothing to do with this. Just note to self: If we end up redesigning the code in this crate, we should consider this.